### ISSUE TITLE
Randomly fails to create google_sql_database_instance due to timing issue

### ISSUE THREAD

**Issue by mogronalol** [hashicorp/terraform#13091]
Hi there,

### Terraform Version
Terraform v0.9.1

### Affected Resource(s)
- google_sql_database_instance

### Terraform Configuration Files
```hcl

provider "google" {
  credentials = "${file("account.json")}"
  region      = "europe-west-1"
  project = "${data.terraform_remote_state.project.project_id}"
}

resource "google_sql_database_instance" "master" {
  region = "europe-west1"
  project = "${data.terraform_remote_state.project.project_id}"
  database_version = "MYSQL_5_7"

  settings {
    tier = "db-n1-standard-1"
    activation_policy = "ALWAYS"
  }
}

data "terraform_remote_state" "project" {
  backend = "gcs"
  config {
    path = "projects/dev.tfstate"
    bucket = "application-cloud-tf-state"
  }
}
```

### Debug Output
https://gist.github.com/mogronalol/c8c308e17b390023939d85ba9c5853a3

### Expected Behavior
Successfully retrieve the instance creation operation, then block until it is complete.

### Actual Behavior
A `404` because it does a `GET` too quickly, so the operation does not exist yet.
https://github.com/hashicorp/terraform/blob/bfdeae0e3357f50a248806c0ff79c85f37215f79/builtin/providers/google/resource_sql_database_instance.go#L570-L570 is the line which fails.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

This does not always fail. It will pretty much never fail if you add a sleep before https://github.com/hashicorp/terraform/blob/bfdeae0e3357f50a248806c0ff79c85f37215f79/builtin/providers/google/resource_sql_database_instance.go#L570-L570 then it doesn't fail. 

What does not make sense, is that if I change `SqlAdminOperationWater.RefreshFunction` to retry on a `404` instead of failing, it will still be a `404` after five attempts. But, a simple sleep before calling `RefreshFunc` for the first time means that there is no `404` so I'm a little stumped.

**Comment by paddyforan** [comment]
Hey @mogronalol! Thanks for reporting the bug. I've been chasing this for the past few weeks (#12436 was written specifically to help track this down) so it's always good to have an extra datapoint. I'm pretty sure this is an upstream bug, and I'll work with Google to get it addressed. (Opening a bug on https://issuetracker.google.com is on my to-do list.)

If you don't mind, I'd love to know how many SQL instances are defined in your config. (Does it reliably work on the config you provided? Or only sometimes?) I have a hunch that the error is more likely to occur the more instances you define in your config, but can't corroborate that yet.

Definitely interested in getting this resolved, either upstream, through a workaround on our end, or both.

**Comment by mogronalol** [comment]
I only  have a single instance defined.

One thing I've noticed since raising this is that if I manually create the project my template works fine. But, if the project is created with Terraform, I get this weird race condition. 

I know that sounds strange but I am wondering if project generation with the upstream API is producing some sort of invalid project state. 

It's just strange because I spent a good part of today implementing a fix by retrying on a 404, but that unfortunately did not work. 

It's really painful as these timing bugs are always hard to diagnose. Adding print statements to try and figure it out was impossible because a failure wasn't deterministic.
